main.go

```diff
@@ -82,5 +94,20 @@ func main() {
 			fmt.Println("resolved outer promise")
 		})
 
+		GetUserNameWithPanic(3).Then(func(x interface{}) {
+			fmt.Println("7 : user:", x)
+		}).Catch(func(err error) {
+			fmt.Println("7 : err:", err)
+		})
+
+		syncResult2, err = GlobalEventLoop.Await(GetUserNameWithPanic(1))
+		if err != nil {
+			fmt.Println(err)
+		}
+		fmt.Println("1 : user:", syncResult2)
+
+		// GetUserNameWithPanic(1).Then(func(x interface{}) {
+		// 	fmt.Println("7 : user:", x)
+		// })
```

Comment:
Remove this block, I don't think it's needed, the above examples are enough to convey the information 👍.